contracts/pools/weighted/WeightedPool2Tokens.sol

```diff
@@ -736,7 +748,7 @@ contract WeightedPool2Tokens is
     // Oracle functions
 
     function _updateOracle(uint256 lastChangeBlock) internal {
-        if (_isOracleEnabled() && block.number > lastChangeBlock) {
+        if (isOracleEnabled() && block.number > lastChangeBlock) {
             bytes32 miscData = _miscData;
```

Comment:
```suggestion
        bytes32 miscData = _miscData;
        bool oracleEnabled = _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
        if (oracleEnabled() && block.number > lastChangeBlock) {
```

Seems a bit silly to read _miscData twice, no? Regardless, we can ignore this now, and focus on performance once all parts are ready.

Reply:
Yea, I did this initially without taking into account gas reads/writes optimizations for this variable in particular... I have a whole plan to refactor the way we access to it entirely once the integration is finished. I wouldn't spend to much time about it rn.